site/source/docs/debugging/Sanitizers.rst

```diff
@@ -0,0 +1,51 @@
+.. _Sanitizers:
+
+=========================
+Debugging with Sanitizers
+=========================
+
+Undefined Behaviour Sanitizer
+=============================
+
+Clang's `undefined behavior sanitizer`__ (UBSan) is now available for use with
+Emscripten. This makes it much easier to catch bugs in your code.
+
+__ https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
+
+To use UBSan, simply pass ``-fsanitize=undefined`` to ``emcc`` or ``em++``.
+
+Catching Null Dereference
+-------------------------
+
+With Emscripten, deferencing a null pointer does not immediately cause a
+segmentation fault, unlike traditional platforms. Instead, Emscripten checks
+at the end of the program execution. This makes is difficult to find where the
+null dereference occurred.
+
+UBSan will tell you exactly where the null deference happened.
+
+Consider the following program, ``null-assign.c``:
+
+.. code-block:: c
+
+  int main(void) {
+      int *a = 0;
+      *a = 0;
+  }
+
+Without UBSan, you get an error when the program exits:
+
+.. code-block:: console
+
+  $ emcc null-assign.c
+  $ node a.out.js
+  Runtime error: The application has corrupted its heap memory area (address zero)!
```

Comment:
Does this only happen if you write to address 0? If so, it would be good to mention that this only catches reads.

Reply:
Yes, emscripten by default can only catch null pointer writes. This is clarified in the next version.